granulate_utils/metadata/databricks_client.py

```diff
@@ -43,46 +50,80 @@ def get_webui_address() -> Optional[str]:
         # The API used: https://spark.apache.org/docs/latest/monitoring.html#rest-api
         apps_url = SPARKUI_APPS_URL.format(webui)
-        self.logger.debug(f"Databricks SparkUI address: {apps_url}.")
-        resp = requests.get(apps_url, timeout=REQUEST_TIMEOUT)
-        if not resp.ok:
-            self.logger.warning(
-                f"Failed initializing Databricks client. {apps_url!r} request failed, status_code: {resp.status_code}."
-            )
+        self.logger.debug(f"Databricks SparkUI address: {apps_url}")
+        try:
+            response = self._request_get(apps_url)
+        except requests.exceptions.RequestException:
+            # Request might fail in cases where the cluster is still initializing, retrying.
             return None
-        apps = resp.json()
+        try:
+            apps = response.json()
+        except Exception as e:
+            raise DatabricksJobNameDiscoverException(f"Failed to parse apps url response, query {response=}") from e
         if len(apps) == 0:
-            self.logger.warning("Failed initializing Databricks client. There are no apps.")
+            # apps might be empty because of initialization, retrying.
+            self.logger.debug("No apps yet, retrying.")
             return None
-        # There's an assumption that only one app exists, and even if there are more -
-        # the name of the job should be the same.
-        env_url = f"{apps_url}/{apps[0]['id']}/environment"
-        resp = requests.get(env_url, timeout=REQUEST_TIMEOUT)
-        if not resp.ok:
-            self.logger.warning(
-                f"Failed initializing Databricks client. {env_url!r} request failed, status_code: {resp.status_code}."
-            )
-            return None
-        env = resp.json()
-        props = env["sparkProperties"]
+
+        env_url = f"{apps_url}/{apps[0].get('id')}/environment"
+        try:
+            response = self._request_get(env_url)
+        except Exception as e:
+            # No reason for any exception, `environment` uri should be accessible if we have running apps.
+            raise DatabricksJobNameDiscoverException(f"Environment request failed {env_url=}") from e
+        try:
+            env = response.json()
+        except Exception as e:
+            raise DatabricksJobNameDiscoverException(f"Environment request failed {response=}") from e
+        props = env.get("sparkProperties", [])
+        if not props:
+            raise DatabricksJobNameDiscoverException(f"sparkProperties was not found in {env=}")
         for prop in props:
             if prop[0] == CLUSTER_TAGS_KEY:
-                for tag in json.loads(prop[1]):
-                    if tag["key"] == JOB_NAME_KEY:
-                        return str(tag["value"])
+                try:
+                    all_tags_value = json.loads(prop[1])
+                except Exception as e:
+                    raise DatabricksJobNameDiscoverException(f"Failed to parse {prop=}") from e
```

Comment:
The various try-except blcoks do not make sense in this function 🤷
One try-except block on the outside should give you all the debug info you need.